test/suites/host-ip-change.robot

```diff
@@ -0,0 +1,233 @@
+*** Settings ***
+Documentation       Host IP and Certificate change test Suite
+
+Resource            ../resources/common.resource
+Resource            ../resources/systemd.resource
+Resource            ../resources/microshift-config.resource
+Resource            ../resources/microshift-process.resource
+
+Library             OperatingSystem
+Library             SSHLibrary
+Library             BuiltIn
+Library             String
+Library             random
+
+Suite Setup         Setup Suite
+Suite Teardown      Teardown Suite
+
+Test Tags           restart    slow
+
+*** Variables ***
+${HOSTS_FILE_PATH}  /etc/hosts
+${EXTERNAL_SERVER_CERT}    /var/lib/microshift/certs/kube-apiserver-external-signer/kube-external-serving/server.crt
+
+*** Test Cases ***
+Verify Certificate Updates During Host IP Change
+    [Documentation]  Ensure certificate updates when the host IP changes
+    # Generate new IP in same subnet
+    Generate New IP
+
+    # Take backup of /etc/hosts and /etc/NetworkManager/system-connections/8.nmconnection
+    ${hosts_backup}=  Save Backup Default File    ${HOSTS_FILE_PATH}
+    Log    Hosts file backed up as: ${hosts_backup}
+
+    ${nmconnection_backup}=  Save Backup Default File    ${NMCONNECTION_PATH}
+    Log    ens3.nmconnection backed up as: ${nmconnection_backup}
+
+    # Take initial checksum of the external server cert ffile
+    # Save the initial checksum of the file
+    ${initial_checksum}=    Get File Checksum    ${EXTERNAL_SERVER_CERT}
+
+    # Add node name and IP to /etc/hosts
+    ${node_name}=    Get NodeName
+    Add Node To Hosts File    ${NEW_HOST_IP}    ${node_name}
+
+    # Stop MicroShift
+    Stop MicroShift
+
+    # Modify the NetworkManager configuration using nmcli commands
+    Modify NetworkManager Configuration   ${NEW_HOST_IP}/24   ${GATEWAY}   ${DEVICE}
+
+    # Reboot the MicroShift system vm
+    Reboot MicroShift System
+    Sleep   200s
+
+    # Login to the MicroShift host with IP via ssh and leave the connection open
+    Login MicroShift Host With New IP    ${NEW_HOST_IP}
+
+    # Verify changed IP in Microshift config
+    Verify Microshift Show Config Command Output   ${NEW_HOST_IP}
+
+    # Verify new Kubeconfig generated for new IP
+    Check New Kubeconfig Generated For New IP
+
+    # Compare the checksum of the external server cert file after the operations
+    Compare File Checksum    ${EXTERNAL_SERVER_CERT}    ${initial_checksum}
+
+    # Rollback configuration after the test case completes
+    Rollback Configuration    ${HOSTS_FILE_PATH}
+    Rollback Configuration    ${NMCONNECTION_PATH}
+
+    # Reboot the MicroShift system vm after rollback
+    Reboot MicroShift System
+    Sleep   200s
+   
+    # Verify microshift cluster work fine after rollback
+    Get NodeName
```

Comment:
How does fetching the node name verify that the cluster is working?

One thing we saw failing in your previous testing was `oc exec` or `oc debug node` calls. Can we use one of those to run a command to verify that the system is working?

Deploying a service would also be a good test, maybe there are parts of another suite that can be reused for that here.

Reply:
Done